test/helpers/types.go

```diff
@@ -99,3 +102,102 @@ func (l logs) NonEmpty() bool {
 	}
 	return len(l) > 0
 }
+
+func isNil(i interface{}) bool {
+	if i == nil {
+		return true
+	}
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
+		return reflect.ValueOf(i).IsNil()
+	}
+	return false
+}
+
+func DeepFields(iface interface{}, namePrefix string) ([]reflect.Value, []string) {
+	fields := make([]reflect.Value, 0)
+	names :=  make([]string,0)
+
+	ifv := reflect.ValueOf(iface)
+	ift := reflect.TypeOf(iface)
+
+	for i := 0; i < ift.NumField(); i++ {
+		v := ifv.Field(i)
+		n := namePrefix + ifv.Type().Field(i).Name
+
+		switch v.Kind() {
+		case reflect.Struct:
+			moreFields, moreNames := DeepFields(v.Interface(), n + "_")
+			fields = append(fields, moreFields...)
+			names = append(names, moreNames...)
+		case reflect.Ptr:
+			if !isNil(v.Interface()) {
+				elm := v.Elem()
+				moreFields, moreNames := DeepFields(elm.Interface(), n + "_")
+				fields = append(fields, moreFields...)
+				names = append(names, moreNames...)
+			}
+		default:
+			fields = append(fields, v)
+			names = append(names, n)
+		}
+	}
+
+	return fields, names
+}
+
+func compareLogLogic (name string, templateValue interface{}, value interface{}) bool {
+	if templateValue == value { // Same value is ok
+		logger.V(3).Info("CompareLogLogic: Same value for", "name", name, "value", templateValue)
+		return true
+	}
+	if templateValue == "*" && !isNil(value) { // Any value, not Nil is ok if template value is "*"
+		logger.V(3).Info("CompareLogLogic: Any value for * ", "name", name, "value", value)
+		return true
+	}
+	if strings.HasPrefix(templateValue.(string), "Pre-") { // Any prefix value is ok if template value starts with "Pre-"
+		if strings.HasPrefix(value.(string)[4:], templateValue.(string)[4:])  {
+			logger.V(3).Info("CompareLogLogic: Fit prefix ", "name", name, "value", value)
+			return true
+		}
+	}
+
+	return false
+}
+
+func (output Log) CompareLog(input Log) bool {
```

Comment:
We should add tests to exercise this code

Reply:
Done